src/coreclr/utilcode/clrhost_nodependencies.cpp

```diff
@@ -204,96 +204,31 @@ ClrDebugState *CLRInitDebugState()
 
 // use standard heap functions for AddressSanitizer and for the DAC build.
 #if !defined(HAS_ADDRESS_SANITIZER) && !defined(DACCESS_COMPILE)
-#ifdef _DEBUG
-#ifdef TARGET_X86
-#define OS_HEAP_ALIGN 8
-#else
-#define OS_HEAP_ALIGN 16
-#endif
-#define CLRALLOC_TAG 0x2ce145f1
-#endif
-
-#ifdef HOST_WINDOWS
-static HANDLE g_hProcessHeap;
-#endif
 
-static FORCEINLINE void* ClrMalloc(size_t size)
+#if defined(SELF_NO_HOST)
+namespace
 {
-    STATIC_CONTRACT_NOTHROW;
-
-#ifdef FAILPOINTS_ENABLED
-    if (RFS_HashStack())
-        return NULL;
-#endif
-
-    void* p;
-
-#ifdef _DEBUG
-    size += OS_HEAP_ALIGN;
-#endif
-
-#ifdef HOST_WINDOWS
-    HANDLE hHeap = g_hProcessHeap;
-    if (hHeap == NULL)
+    void ReportOOM(size_t size)
     {
-        InterlockedCompareExchangeT(&g_hProcessHeap, ::GetProcessHeap(), NULL);
-        hHeap = g_hProcessHeap;
+        // With no host, we have nowhere to report the OOM.
     }
-
-    p = HeapAlloc(hHeap, 0, size);
```

Comment:
How much more expensive is malloc compared to HeapAlloc on Windows?

Reply:
Nowadays, `malloc` just calls `HeapAlloc`, so it should be minimal extra cost. Maybe a few instructions for a tail call or a well-predicted if check (checking if we're using the system heap, which is the default).

Reply:
I have done a quick perf check on Winx64 and I do not see a measurable impact from this change.